velox/functions/prestosql/types/JsonType.cpp

```diff
@@ -856,30 +856,46 @@ struct CastFromJsonTypedImpl {
         }
       } else {
         SIMDJSON_ASSIGN_OR_RAISE(auto object, value.get_object());
-        folly::F14FastMap<std::string, simdjson::ondemand::value> lowerCaseKeys(
-            object.count_fields());
+
+        // TODO Populate this mapping once, not per-row.
+        // Mapping from lower-case field names of the target RowType to their
+        // indices.
+        folly::F14FastMap<std::string, column_index_t> fieldIndices;
+        for (auto i = 0; i < rowType.size(); ++i) {
+          auto key = rowType.nameOf(i);
+          boost::algorithm::to_lower(key);
+          fieldIndices[key] = i;
+        }
+
+        std::vector<bool> foundIndices(rowType.size(), false);
```

Comment:
We could initialize `fieldIndices` value to -1 then we don't need this second memory.

Reply:
Sounds good. Updated.